packages/capabilities/test/capabilities/provider.test.js

```diff
@@ -0,0 +1,471 @@
+            with: agent.did(),
+          }),
+        ],
+      })
+
+      const result = await access(await auth.delegate(), {
+        capability: Provider.get,
+        principal: Verifier,
+        authority: service,
+      })
+      if (result.error) {
+        assert.fail('error in self issue')
+      } else {
+        assert.deepEqual(result.audience.did(), service.did())
+        assert.equal(result.capability.can, 'provider/get')
+        assert.deepEqual(result.capability.nb, {
+          provider: 'did:web:ucan.web3.storage:providers:free',
+          consumer: space.did(),
+        })
+      }
+    })
+
+    it('should invoke from session', async function () {
+      const agent = alice
+      const space = bob
+      const account = agent.withDID('did:mailto:alice@mail.com')
+      const auth = Provider.get.invoke({
+        issuer: account,
+        audience: service,
+        with: account.did(),
+        nb: {
+          provider: 'did:web:ucan.web3.storage:providers:free',
+          consumer: space.did(),
+        },
+        proofs: [
+          await Access.session
+            .invoke({
+              issuer: service,
+              audience: account,
+              with: service.did(),
+              nb: {
+                key: agent.did(),
+              },
+            })
+            .delegate(),
+        ],
+      })
+
+      const result = await access(await auth.delegate(), {
+        capability: Provider.get,
+        principal: Verifier,
+        authority: service,
+      })
+      if (result.error) {
+        assert.fail('error in self issue')
+      } else {
+        assert.deepEqual(result.audience.did(), service.did())
+        assert.equal(result.capability.can, 'provider/get')
+        assert.deepEqual(result.capability.nb, {
+          provider: 'did:web:ucan.web3.storage:providers:free',
+          consumer: space.did(),
+        })
+      }
+    })
+
+    it('should fail using account issuer without session', async function () {
+      const agent = alice
+      const space = bob
+      const account = agent.withDID('did:mailto:alice@mail.com')
+      const auth = Provider.get.invoke({
+        issuer: account,
+        audience: service,
+        with: account.did(),
+        nb: {
+          provider: 'did:web:ucan.web3.storage:providers:free',
+          consumer: space.did(),
+        },
+      })
+
+      const result = await access(await auth.delegate(), {
+        capability: Provider.get,
+        principal: Verifier,
+        authority: service,
+      })
+
+      if (result.error) {
+        assert.ok(
+          result.message.includes(`Unable to resolve '${account.did()}' key`)
+        )
+      } else {
+        assert.fail('should return error')
+      }
+    })
+
+    it('should fail session key does not match invocation signature', async function () {
+      const agent = alice
+      const space = bob
+      const account = agent.withDID('did:mailto:alice@mail.com')
```

Comment:
```suggestion
      const account = agent.withDID('did:mailto:mail.com:alice')
```

let's use valid did syntax in the tests

Reply:
I made one other suggestion like this, but avoided doing it everywhere in case there's a good reason to use the syntax with `@` in the immediate term